### DISCUSSION TITLE
Working Example of Parallel Processing?

### DISCUSSION THREAD

**Discussion by PeteMichaud** [SixLabors/ImageSharp#2046] [Ideas]
I have had no luck at all searching google or github for a working example of a `ParallelRowIterator` in action. My goal is to loop through each pixel of a very large image and set all the values to something. How best to parallelize the operation? This would be great to add to the docs somewhere!

**Comment by PeteMichaud** [comment]
I'm trying to do something like:

```
Image<Rgb24> sourceImage = GetImage();
Image<Rgb24> outImage = GetImage();

for (var outY = 0; outY < outImage.Height; outY++)
{
    for (int outX = 0; outX < outImage.Width; outX++)
    {
        var outColor = GetArbitraryPixelFromAnywhereInsideSourceImage(sourceImage);
         outImage[outX, outY] = outColor;
    }
}
```

The main thing confusing me is the arbitrary access from the sourceImage. I can go row by row, pixel by pixel in the outImage, but I don't know what sourceImage row or column I'll need until I get inside this loop, and I will most of the time need to access not only a target pixel, but a sample of many of the pixels around it (eg. bicubic sampling).

I have working code that does this processing, but I can't figure out how to use the parallel access api to parallelize it, and just using Parallel.For scrambles the outImage (which I'm a little confused about actually, since each outPixel is an independent calculation and the sourceImage never changes).

**Reply by JimBobSquarePants** [reply]
I want to highlight here that you are misusing the Ideas discussion forum for questions.

![image](https://user-images.githubusercontent.com/385879/157134591-47653bce-aced-411c-8a2c-1b405fa649e3.png)

There is no help category by design. This software is released [AS IS](https://github.com/SixLabors/ImageSharp/blob/1859d0f1bcb012083ab6e55386bd9e3c0b291e2f/LICENSE#L143-L151) unless you [purchase a Six Labors License](https://sixlabors.com/pricing/). Alternatively you can ask on [Stack Overflow](https://stackoverflow.com/questions/tagged/imagesharp) and maybe a community member will prvide an answer.

**Comment by PeteMichaud** [comment]
Sorry @JimBobSquarePants, my mistake. I just came here from the link in the [top level readme](https://github.com/SixLabors/ImageSharp) where it says:

> Do you have questions? We are happy to help! Please [join our Discussions Forum](https://github.com/SixLabors/ImageSharp/discussions/category_choices), or ask them on [Stack Overflow](https://stackoverflow.com/) using the ImageSharp tag. Please do not open issues for questions.

So I thought it was the right place.

**Reply by JimBobSquarePants** [reply]
Our bad, I've updated the readme. I'll answer over at SO

**Comment by PeteMichaud** [comment]
Anyway, I posted the question here for future reference: https://stackoverflow.com/questions/71388492/right-way-to-parallelize-pixel-access-across-multiple-images-using-imagesharp
